libraries/aws/provisioning/src/private/aws_iot_provisioning_internal.h

```diff
@@ -522,6 +585,22 @@ AwsIotProvisioningError_t _AwsIotProvisioning_ParseRegisterThingResponse( AwsIot
 AwsIotProvisioningError_t _AwsIotProvisioning_SerializeCreateKeysAndCertificateRequestPayload( uint8_t ** pSerializationBuffer,
                                                                                                size_t * pBufferSize );
 
+/**
+ * @brief Serializes payload data in a buffer, if provided, of a request to the CreateCertificateFromCsr API on AWS IoT Core;
+ * otherwise stores the calculated size of the serialized data in the passed parameter, @a pBufferSize.
+ *
+ * @param[in] pCertificateSigningRequest The Certificate-Signing Request string to serialize for the request.
+ * @param[in] csrLength The length of the Certificate-Signing Request string.
+ * @param[in] pSerializationBuffer The buffer for storing the serialized payload data.
+ * @param[in, out] pBufferSize If the payload buffer is provided, this should contain the size of the buffer. Otherwise,
```

Comment:
> What is the return value if pSerializationBuffer is NULL?
That is the case for a dry-run of serialization for calculating the required buffer size. The `pBufferSize` output parameter will be populated with the calculated size

> What about if pSerializationBuffer is non-NULL and pBufferSize is too small?
It will return an error as the return value of the call.

Do you prefer that I break down these use-cases in 2 different functions instead?

Reply:
Or at least define and document those two behaviors.

Reply:
Separated the 2 use-cases in different functions `_AwsIotProvisioning_CalculateCertFromCsrPayloadSize` and `_AwsIotProvisioning_SerializeCreateCertFromCsrRequestPayload` in [this commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/826/commits/cb41a55110f582211428e5085505223a95d50fad#diff-77f76cfbe42e8353b432bae257edbe4e)